src/fairseq2/recipes/hg/asr_eval.py

```diff
@@ -17,44 +18,86 @@
 )
 
 from fairseq2.config_registry import ConfigRegistry
-from fairseq2.data.data_pipeline import SequenceData
 from fairseq2.data.text import load_text_tokenizer
 from fairseq2.data.text.text_tokenizer import TextTokenizer
 from fairseq2.datasets.batching import StaticBatching
 from fairseq2.logging import get_log_writer
-from fairseq2.models.seq2seq import Seq2SeqBatch
 from fairseq2.models.sequence import SequenceBatch
 from fairseq2.models.wav2vec2.asr import load_wav2vec2_asr_model
-from fairseq2.nn.padding import get_seqs_and_padding_mask
+from fairseq2.nn.padding import PaddingMask, get_seqs_and_padding_mask
 from fairseq2.recipes.hg.dataset import Example, create_hf_reader
 from fairseq2.recipes.hg.evaluator import HFEvaluator
 from fairseq2.recipes.utils.setup import setup_root_gang
-from fairseq2.typing import META, DataType
+from fairseq2.typing import META, DataType, Device
 from fairseq2.utils.profiler import Stopwatch
 
 log = get_log_writer(__name__)
 
 
+@dataclass
+class AsrDatasetConfig:
+    """Configuration for an automatic speech recognition dataset."""
+
+    dataset_path: str
+    """The name of the dataset."""
+
+    dataset_name: Optional[str] = None
+    """The name of the dataset split."""
+
+    split: str = "test"
+    """The name of the dataset split to evaluate with."""
+
+    source_column: List[str] = field(default_factory=list)
+    """The path of the column containing the source audio."""
+
+    target_column: List[str] = field(default_factory=list)
+    """The path of the column containing the target text."""
+
+    @classmethod
+    def from_dict(cls, config_dict: dict[str, Any]) -> "AsrDatasetConfig":
+        """Create an AsrDatasetConfig instance from a configuration dictionary."""
+        return cls(
+            dataset_path=config_dict.get("dataset_path", ""),
+            dataset_name=config_dict.get("dataset_name"),
+            source_column=config_dict.get("source_column", []),
+            target_column=config_dict.get("target_column", []),
+            split=config_dict.get("split", "test"),
+        )
+
+    def get_source_data(self, ds: Example) -> Any:
```

Comment:
I would prefer `get_source_data` and `get_target_data` to be standalone functions (along with `_get_data`), instead of being part of the config type. A configuration dataclass should ideally contain only fields and functions (e.g. `from_dict`) to construct instances.

Reply:
The `get_source_data` and `get_target_data` functions are essentially wrappers around `_get_data` to simplify its usage. Removing these functions from the class would eliminate their purpose. Instead, I can refactor the code to move `_get_data` outside the class and use it directly.

Do you want me to do this?

Reply:
In https://github.com/facebookresearch/fairseq2/pull/749/commits/667194fd968c82f46ff43e85aaf57f13127e3be7, I have moved the `_get_data` outside of the dataclass, renamed it to `_get_column_data`, removed both `get_source_data` and `get_target_data` and updated the code to use it instead.

I will mark this as resolved